.github/workflows/build_and_test_rpms.yml

```
@@ -0,0 +1,103 @@
+name: Build and test blahp RPMs
+on: [pull_request, push]
+
+jobs:
+  build-rpms-and-test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        dver: [7]
+        target_env: [osg, uw_build]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Prepare Docker
+        run: |
+          echo 'DOCKER_OPTS="-H tcp://127.0.0.1:2375 -H unix:///var/run/docker.sock -s devicemapper"' | sudo tee /etc/default/docker > /dev/null &&
+            sudo service docker restart
+
+      - name: Start CentOS ${{ matrix.dver}} image
+        run: |
+          docker run --privileged --detach --env "container=docker" \
+                 --hostname "blahp-ci.htcondor.org" \
+                 --name $GITHUB_SHA \
+                 --volume /sys/fs/cgroup:/sys/fs/cgroup \
+                 --volume `pwd`:/blahp:rw \
+                 centos:centos${{ matrix.dver }} \
+                 /usr/sbin/init
+
+      - name: Build EL${{ matrix.dver }} RPMs (${{ matrix.target_env }})
+        run: |
+          docker exec $GITHUB_SHA \
+                 /bin/bash -xc \
+                   "/blahp/tests/build_rpms.sh \
+                     ${{ matrix.dver }} \
+                     ${{ matrix.target_env}}"
+
+      - name: Install EL${{ matrix.dver }} RPMs (${{ matrix.target_env }})
+        run: |
+          docker exec $GITHUB_SHA \
+                 /bin/bash -xc \
+                   "/blahp/tests/install_rpms.sh \
+                     ${{ matrix.dver }} \
+                     ${{ matrix.target_env}}"
+
+      - name: Run ${{ matrix.target_env }} integration tests (EL${{ matrix.dver }})
+        id: integration-tests
+        run: |
+          docker exec $GITHUB_SHA \
+                 /bin/bash -xc \
+                   "/blahp/tests/test_inside_docker.sh \
+                     ${{ matrix.dver}} \
+                     ${{ matrix.target_env}}"
+
+      - name: Host certificate details
+        if: failure() && steps.integration-tests.outcome == 'failure'
+        run: |
+          docker exec $GITHUB_SHA \
+                 openssl x509 -in /etc/grid-security/hostcert.pem -noout -text
+
+      - name: HTCondor configuration details
+        if: failure() && steps.integration-tests.outcome == 'failure'
+        run: |
+          docker exec $GITHUB_SHA \
+                 condor_config_val -dump
+
+      - name: Slurm logs
+        if: failure() && steps.integration-tests.outcome == 'failure'
+        run: |
+          docker exec $GITHUB_SHA \
+                 /bin/bash -c "cat /var/log/slurm/*"
+
+      - name: HTCondor-CE logs
+        if: failure() && steps.integration-tests.outcome == 'failure'
+        run: |
+          docker exec $GITHUB_SHA \
+                 /bin/bash -c "find /var/log/condor-ce -type f -exec cat {} \;"
```

In general for these commands, is the idea just to test if the command runs successfully, or do you actually want to capture the output for later inspection?

If it's the latter, note that running `cat` on a bunch of files will not show which lines belong to which files in the output.  (You could either add a header/footer with additional items in the `find` command (eg with `-printf`), or else do something like `grep ^` instead of `cat`.)

Also, the `bash -c` thing here is fine, and is kind of necessary when you want a glob (`*`) to be evaluated inside the container rather than where the docker command line is invoked; it seems in the case of this `find` command line (with no wildcards), i _think_ you can also get away with just

```suggestion
                 find /var/log/condor-ce -type f -exec cat {} \;
```

(wild, right?)

Same goes for every other `docker exec ... bash -c "command line"`, where `command line` doesn't have any wildcards.

---

We want the actual output for inspection in case of test failure. I've kicked off another branch that tries out your suggested change

---

The change works. It would be nice to have separators between files but we're in a bit of a rush to get this in so I'm going to go ahead and merge